### PR TITLE
rustfmt fixes

### PR DESCRIPTION
**Pull request by nnethercote**
The `rmake.rs` entries in `rustfmt.toml` are causing major problems for `x fmt`. This PR removes them and does some minor related cleanups.

r? @GuillaumeGomez

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -3,25 +3,18 @@ version = "Two"
...
     "/build/",
     "/*-build/",
     "/build-*/",
     "/vendor/",
 
-    # tests for now are not formatted, as they are sometimes pretty-printing constrained
-    # (and generally rustfmt can move around comments in UI-testing incompatible ways)
-    "/tests/*",
+    # Tests for now are not formatted, as they are sometimes pretty-printing constrained
+    # (and generally rustfmt can move around comments in UI-testing # incompatible ways).
```
Comment (lqd) [inline rustfmt.toml:14]: Stray # I presume

**Comment by GuillaumeGomez** [comment]
Instead of removing the negated entries, wouldn't it be better to actually fix our bootstrap is handling this? It's very weird after all...

**Comment by GuillaumeGomez** [comment]
Just noting: if you don't have time to do it, r=me for this PR so it unlocks the situation and open an issue about (and ping me into it) so I'll fix it once I have time.

**Comment by nnethercote** [comment]
In short, the problem is that `ignore` does some quite strange things. If you whitelist zero files in `OverrideBuilder` it will traverse all files implicitly (except those that are explicitly ignored). But if you whitelist a single file, no files are implicitly traversed. It's a design/API issue with `ignore`, and seems hard to work around.

File: rustfmt.toml (hunk 1 above)
Comment (nnethercote) [reply rustfmt.toml]: Fixed, thanks.

**Comment by GuillaumeGomez** [comment]
I don't think we should use `OverrideBuilder` but [`GitignoreBuilder`](https://docs.rs/ignore/latest/ignore/gitignore/struct.GitignoreBuilder.html) instead and actually write a correct `rustfmt.toml` file.

**Comment by nnethercote** [comment]
Anyway, let's unbreak things. I do have some follow-up work that will come in another PR.

@bors r=GuillaumeGomez rollup

**Comment by GuillaumeGomez** [comment]
`rustfmt` relies on them and things seem to work as expected overall. Having an entire set of rust files not following the same formatting convention is a worse outcome in my opinion. Well, for now it's better to unbreak things, let's continue this work after it.

**Comment by nnethercote** [comment]
> rustfmt relies on them and things seem to work as expected overall.

Until some `!`-prefixed entries were added and things broke horribly. We can grow the set of files that are being formatted, we just have to do it without `!` entries.

**Comment by GuillaumeGomez** [comment]
The problem here is how we handle the `rustfmt.toml` file, not the `!` entries. Anyway, you seem to have ideas on how to fix things. Can't wait to see. :)

**Pull request merged**